sentinelsat/sentinel.py

```diff
@@ -73,6 +75,11 @@ def __init__(self, user, password, api_url='https://scihub.copernicus.eu/apihub/
         # For unit tests
         self._last_query = None
         self._last_response = None
+        # keep track of failed downloads
+        self.failed_list_id = list()
```

Comment:
You add a lot of state to the API here, that is, some lists that will keep growing over subsequent usage of the instance. Better to leave the tracking of failed downloads to the user.

Reply:
sure, I leave them to the users.